tests/validation/tests/v3_api/test_certificate.py

```diff
@@ -0,0 +1,571 @@
+        """
+        Test steps:
+        1. Create an ingress pointing to self signed certificate scoped to
+        all namespace.
+        2. Update the certificate key to trusted.
+        3. Reload the certificate.
+        4. Update the ingress.
+        5. validate the ingress using endpoint.
+        """
+        ingress_name = random_test_name("ingress-test")
+        host_1 = route_entry_53_2
+        host_2 = route_entry_53_1
+        path = "/name.html"
+        rule_1 = {"host": host_1,
+                  "paths": [{"path": path, "workloadIds": [self.workload.id],
+                             "targetPort": "80"}]
+                  }
+        rule_2 = {"host": host_2,
+                  "paths": [{"path": path, "workloadIds": [self.workload.id],
+                             "targetPort": "80"}]
+                  }
+        tls = {"certificateId": self.certificate_all_ns_ssc.id,
+               "hosts": [host_1]}
+        tls_2 = {"certificateId": self.certificate_all_ns_ssc.id,
+                 "hosts": [host_2]}
+        self.ingress = self.p_client.create_ingress(
+            name=ingress_name, namespaceId=self.ns.id, rules=[rule_1],
+            tls=[tls]
+        )
+        wait_for_ingress_to_active(self.p_client, self.ingress)
+        self.p_client.update(
+            self.certificate_all_ns_ssc, key=rancher_private_key,
+            certs=rancher_cert
+        )
+        self.p_client.reload(self.certificate_all_ns_ssc)
+        self.p_client.update(self.ingress, rules=[rule_2], tls=[tls_2])
+        self.p_client.reload(self.ingress)
+        wait_for_ingress_to_active(self.p_client, self.ingress)
+        validate_ingress_using_endpoint(
+            self.p_client, self.ingress, [self.workload], certcheck=True,
+            is_insecure=True
+        )
+
+    @if_test_rbac
+    @pytest.mark.parametrize("role", rbac_role_list)
+    def test_create_certificate(self, role):
+        """
+        Test steps:
+        1. Create certificate for single and all namespace for all role
+        2. Delete the certificates
+        """
+        token = rbac_get_user_token_by_role(role)
+        project = rbac_get_project()
+        ns = rbac_get_namespace()
+        p_client = get_project_client_for_token(project, token)
+        cert_name_1 = random_test_name("cert-rbac")
+        cert_name_2 = random_test_name("cert-rbac")
+        if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
+            with pytest.raises(ApiError) as e:
+                p_client.create_namespaced_certificate(
+                    name=cert_name_1, key=rancher_private_key,
+                    certs=rancher_cert,
+                    namespaceId=ns['name']
+                )
+                p_client.create_certificate(
+                    name=cert_name_2, key=rancher_private_key,
+                    certs=rancher_cert
+                )
+                assert e.value.error.status == 403
```

Comment:
it is hard to tell is the error coming from  create_namespaced_certificate or create_certificate, or both.